airflow/providers/sftp/sensors/sftp.py

```diff
@@ -54,4 +63,9 @@ def poke(self, context: 'Context') -> bool:
                 raise e
             return False
         self.hook.close_conn()
-        return True
+        if self.newer_than:
+            _mod_time = datetime.strptime(mod_time, '%Y%m%d%H%M%S')
+            _newer_than = make_naive(self.newer_than) if not is_naive(self.newer_than) else self.newer_than
+            return _newer_than <= _mod_time
```

Comment:
I decided to assume a naive comparison and allow the sensor to be configured with an offset aware or unaware newer_than parameter, thoughts on this @eladkal ?

Reply:
I think you shoudl convert all dates to UTC. It's surprisingly hard to find what is the timezone of st_mtime but all signs point at UTC.

Reply:
I agree. Assuming the SFTP site is always on the same time-zone as the airflow instance is weaker than assuming UTC.

I'll make that change.